soc/wch/ch32v00x/soc_irq.S

```diff
@@ -0,0 +1,19 @@
+/* Copyright (c) 2024 Michael Hope
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <offsets.h>
```

Comment:
Is there any chance of this SoC to use `riscv-privileged` to allow drop `soc_irq.S` and `vector.S`?

Reply:
It's not obvious, but is worth a deeper look. The core is machine mode only, and the ISR table (or _isr_handler in the single-handler mode) must be aligned to a 1 KiB boundary.

Reply:
@nzmichaelh @kholia is this someting you are going to look into further?

Reply:
If it's OK, I'd like to punt this one for now. The PR has been in progress since June and this seems to be a nice to have to investigate in a followup.